packages/rocketchat-api/server/v1/groups.js

```diff
@@ -449,23 +445,31 @@ RocketChat.API.v1.addRoute('groups.listAll', { authRequired: true }, {
 RocketChat.API.v1.addRoute('groups.members', { authRequired: true }, {
 	get() {
 		const findResult = findPrivateGroupByIdOrName({ params: this.requestParams(), userId: this.userId });
+		const room = RocketChat.models.Rooms.findOneById(findResult.rid, {fields: {broadcast: 1}});
 
-		if (findResult._room.broadcast && !RocketChat.authz.hasPermission(this.userId, 'view-broadcast-member-list')) {
+		if (room.broadcast && !RocketChat.authz.hasPermission(this.userId, 'view-broadcast-member-list')) {
 			return RocketChat.API.v1.unauthorized();
 		}
 
 		const { offset, count } = this.getPaginationItems();
 		const { sort } = this.parseJsonQuery();
 
-		let sortFn = (a, b) => a > b;
-		if (Match.test(sort, Object) && Match.test(sort.username, Number) && sort.username === -1) {
-			sortFn = (a, b) => b < a;
+		let subscriptionSort = {};
+
+		if (!Match.test(sort, Object) || !Match.test(sort.username, Number)) {
+			subscriptionSort = {'u.username': 1};
+		} else {
+			subscriptionSort = {'u.username': sort.username};
 		}
 
-		const members = RocketChat.models.Rooms.processQueryOptionsOnResult(Array.from(findResult._room.usernames).sort(sortFn), {
+		const members = RocketChat.models.Subscriptions.findByRoomId(findResult.rid, {
+			fields: {u: 1},
+			sort: subscriptionSort,
 			skip: offset,
 			limit: count
-		});
+		}).fetch()
+			.filter(s => s.u && s.u._id && s.u.username)
```

Comment:
we could use `"u":{$exists:1}..."u._id": {$exists:1} ... "u.username": {$exists:1} `  or not?

Reply:
We could, need to create a new method `findByRoomIdWhenUserUsernameExists` rather than `findByRoomId`